gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/SchemaVersionLockFile.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.dist;
+
+import com.palantir.logsafe.Preconditions;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface SchemaVersionLockFile {
```

Comment:
It'd be worthwhile to make it Jackson serializable, imo. YAML would be a good choice if we want something more human-readable than JSON.

The reason the deps lock files are in a custom lock format is because we copied it over from nebula. It also wasn't meant to be deserialized and consumed in tooling (at least initially). Here, we're building these files for the purpose of deserializing them, and with no concerns of backwards-compatibility, so having a custom format creates an additional maintenance load that's avoidable. For example, the revapi file, similar to a lock file, is a YAML file: https://github.com/palantir/conjure-java-runtime/blob/develop/.palantir/revapi.yml.

To make that serialization work OOTB with an Immutable, you can add a `@Value.Derived` attribute, `List<SchemaMigrationRange> migrationRanges`, where the derivation logic copies the current `asString()` logic. And that should make it possible to dump the file as a YAML.

Reply:
Okay, makes sense to me now. I wrote it without using `@Value.Derived`